### PR TITLE
finally stop using min/max_value and the integer modules

### PR DESCRIPTION
**Pull request by RalfJung**
https://github.com/rust-lang/rust/pull/68952 landed, so we can finally do this :)

### CODE REVIEW

**Comment by oli-obk** [comment]
Heh, cool!

File: src/shims/foreign_items.rs [hunk 1]
```diff
@@ -443,10 +443,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
...
                 // Saturating cast to i16. Even those are outside the valid exponent range to
                 // `scalbn` below will do its over/underflow handling.
-                let exp = if exp > i16::max_value() as i32 {
-                    i16::max_value()
-                } else if exp < i16::min_value() as i32 {
-                    i16::min_value()
+                let exp = if exp > i16::MAX as i32 {
+                    i16::MAX
+                } else if exp < i16::MIN as i32 {
+                    i16::MIN
```
Comment (eddyb) [inline src/shims/foreign_items.rs]: Hmm I wonder if we can also make similar changes to `rustc_apfloat`. Ugh the licensing mess there isn't sorted out yet, is it...?

Comment (RalfJung) [reply src/shims/foreign_items.rs]: I mean after the hard time you gave me for editing a *comment* in apfloat, I am not touching that code ever again if I can help it.^^

Comment (eddyb) [reply src/shims/foreign_items.rs]: I am not going to touch it either until I hear back about the licensing, don't worry, this was more of a note to myself.

**Pull request merged**